catroid/src/main/java/org/catrobat/catroid/merge/MergeProjects.java

```diff
@@ -0,0 +1,121 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.merge;
+
+import android.content.Context;
+
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.XmlHeader;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.utils.Utils;
+
+public class MergeProjects {
+	private Project firstProject = null;
+	private Project secondProject = null;
+	private Project mergedProject = null;
+	private ConflictHelper conflictHelper = new ConflictHelper();
+	private Context context;
+
+	public MergeProjects(Project firstProject, Project secondProject, Context context) {
+		this.firstProject = firstProject;
+		this.secondProject = secondProject;
+		this.context = context;
+	}
+
+	public MergeProjects(Project firstProject, Project secondProject) {
+		this.firstProject = firstProject;
+		this.secondProject = secondProject;
+	}
+
+	public Project getMergedProject() {
+		return mergedProject;
+	}
+
+	public boolean performMerge(String mergeProjectName) {
+		if (conflictHelper.conflictExist(firstProject, secondProject)) {
+			return false;
+		}
+		mergedProject = new Project(context, mergeProjectName);
+
+		if (mergedProject.getDefaultScene() != null) {
+			mergedProject.removeScene(mergedProject.getDefaultScene());
+		}
+
+		createProjectHeader();
+
+		addResources();
+		XstreamSerializer.getInstance().saveProject(mergedProject);
+
+		return true;
+	}
+
+	private void addResources() {
+		mergedProject.getUserVariables().addAll(firstProject.getUserVariables());
+		mergedProject.getUserVariables().addAll(secondProject.getUserVariables());
+		mergedProject.getUserLists().addAll(firstProject.getUserLists());
+		mergedProject.getUserLists().addAll(secondProject.getUserLists());
+
+		for (Scene scene : firstProject.getSceneList()) {
+			if (!mergedProject.getSceneList().contains(scene)) {
+				mergedProject.getSceneList().add(scene);
+			}
+		}
+
+		for (Scene scene : secondProject.getSceneList()) {
+			if (!mergedProject.getSceneList().contains(scene)) {
+				mergedProject.getSceneList().add(scene);
+			}
+		}
+	}
+
+	private void createProjectHeader() {
+		XmlHeader mainHeader;
+		XmlHeader subHeader;
+		XmlHeader mergeHeader = mergedProject.getXmlHeader();
+
+		int firstProjectSpriteCount = 0;
+		int secondProjectSpriteCount = 0;
+
+		for (Scene scene : firstProject.getSceneList()) {
+			firstProjectSpriteCount += scene.getSpriteList().size();
+		}
+
+		for (Scene scene : secondProject.getSceneList()) {
+			secondProjectSpriteCount += scene.getSpriteList().size();
+		}
+
+		if (firstProjectSpriteCount < 2 && secondProjectSpriteCount > 1) {
```

Comment:
How is this supposed to decide? Is this a specified behavior? If yes where is it specified?

Reply:
I just asked. Wolfgang about it, he said I should use the header for the first project for now until later.